### ISSUE TITLE
Czkawka lists a file in the same folder twice, when it "deduplicates" that the file vanishes.

### ISSUE THREAD

**Issue by ezxpro** [qarmin/czkawka#1187]
**Bug Description**
Upon running this program the first time ever, I noticed it finds "duplicates" that aren't duplicates at all. It lists a file that exists only once as being a duplicate of itself. When you select that for deduplication, your file is gone, causing data loss.

**Steps to reproduce:**
No idea, I just ran this for the first time with the "blake3" hash and it showed several entries where a file is listed as a duplicate of itself, both entries being in the same directory, having the same name. In other words, it points to the same file twice

![image](https://github.com/qarmin/czkawka/assets/15344931/f566d656-5110-4d62-871a-e24e708977a7)

**Comment by qarmin** [comment]
But this paths are different - one have `,` in second `)`

**Comment by ezxpro** [comment]
> But this paths are different - one have `,` in second `)`

Just checked that, you're right, both on the image and on my disk, those folders were there, with slightly different names.

Thanks, sorry for bothering
